### PR TITLE
added modern theme

### PR DESCRIPTION
**Pull request by p01arst0rm**
## Purpose

added modern theme

JIRA issue: [CORE-13020](https://jira.reactos.org/browse/CORE-13020)

## TODO

_Use a TODO when your pull request is Work in Progress._

- more bitmaps
- rewrite textfiles
- finish dark mode

### CODE REVIEW

File: media/themes/modern.msstyles/lang/en-US.rc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+STRINGTABLE
+{
+1000, "Light"
+1001, "Dark"
+}
+
+STRINGTABLE
+{
+2000, "Modern Light"
+2000, "Modern Dark"
```
Comment (learn-more) [inline media/themes/modern.msstyles/lang/en-US.rc]: These have the same ID, which is not allowed.

File: boot/bootdata/packages/reactos.dff.in [hunk 2]
```diff
@@ -78,7 +78,7 @@ Signature = "$ReactOS$"
 58 = winsxs\x86_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.7601.23038_none_deadbeef
 59 = winsxs\x86_reactos.apisets_6595b64144ccf1df_1.0.0.0_none_deadbeef
 60 = winsxs\x86_reactos.newapi_6595b64144ccf1df_1.0.0.0_none_deadbeef
-
+61 = Resources\Themes\Modern
 .InfEnd
```
Comment (learn-more) [inline boot/bootdata/packages/reactos.dff.in]: Please leave a newline between the last entry and the `.InfEnd`

File: media/themes/CMakeLists.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
-
```
Comment (learn-more) [inline media/themes/CMakeLists.txt]: Please leave the first line blank.

File: sdk/cmake/CMakeMacros.cmake [hunk 4]
```diff
@@ -199,7 +199,7 @@ macro(dir_to_num dir var)
         set(${var} 16)
     elseif(${dir} STREQUAL reactos/system32/wbem)
         set(${var} 17)
-    elseif(${dir} STREQUAL reactos/Resources/Themes/Lautus)
+	elseif(${dir} STREQUAL reactos/Resources/Themes/Lautus)
```
Comment (binarymaster) [inline sdk/cmake/CMakeMacros.cmake]: Unnecessary change.

File: sdk/cmake/CMakeMacros.cmake [hunk 5]
```diff
@@ -285,6 +285,8 @@ macro(dir_to_num dir var)
         set(${var} 59)
     elseif(${dir} STREQUAL reactos/winsxs/x86_reactos.newapi_6595b64144ccf1df_1.0.0.0_none_deadbeef)
         set(${var} 60)
+	elseif(${dir} STREQUAL reactos/Resources/Themes/Modern)
```
Comment (binarymaster) [inline sdk/cmake/CMakeMacros.cmake]: Indentation broken (should be 4 spaces instead of tab).

File: media/themes/modern.msstyles/lang/cs-CZ.rc [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+{
+5000, "modern"
+5001, "Motiv"
+5002, "Nadace ReactOS"
+5003, "Pisarz, založeno na Luna Inspirat"
+5004, "Uvolněno pod licencí GNU/GPL 3.0, 2018"
+5005, "http://www.reactos.org"
+5006, "1.0"
+5007, "Motiv pro ReactOS"
+}
```
Comment (learn-more) [inline media/themes/modern.msstyles/lang/cs-CZ.rc]: Please end this file (and all other ones) with an empty line.
https://stackoverflow.com/a/72409

File: media/themes/modern.msstyles/modern.rc [hunk 7]
```diff
@@ -0,0 +1,387 @@
+#define REACTOS_VERSION_DLL
+#define REACTOS_STR_FILE_DESCRIPTION	"Modern Theme\0"
+#define REACTOS_STR_INTERNAL_NAME	"Modern\0"
+#define REACTOS_STR_ORIGINAL_FILENAME	"modern.msstyles\0"
```
Comment (learn-more) [inline media/themes/modern.msstyles/modern.rc]: These 3 lines (2,3,4) should not need the `\0`

File: media/themes/CMakeLists.txt [hunk 8]
```diff
@@ -1,3 +1,3 @@
...
 add_subdirectory(blackshade.msstyles)
 add_subdirectory(lautus.msstyles)
+add_subdirectory(modern.msstyles)
```
Comment (binarymaster) [inline media/themes/CMakeLists.txt]: Case sensitive inconsistence problem (`modern.msstyles` vs. `Resources\Themes\Modern`).

File: media/themes/modern.msstyles/lang/en-US.rc [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+4001, "Large Fonts"
+4002, "Extra Large Fonts"
+}
+
+STRINGTABLE
+{
+5000, "Modern"
+5001, "Visual Style"
+5002, "ReactOS Foundation"
+5003, "Pisarz, based on Luna Inspirat"
```
Comment (learn-more) [inline media/themes/modern.msstyles/lang/en-US.rc]: `5003` means author. You could also add (just like the author from lautus did) indicate where you based it off, which would be lautus in this case.

File: sdk/cmake/CMakeMacros.cmake (hunk 4 above)
Comment (learn-more) [inline sdk/cmake/CMakeMacros.cmake]: We use spaces, you inserted a tab here.

File: sdk/cmake/CMakeMacros.cmake (hunk 5 above)
Comment (learn-more) [inline sdk/cmake/CMakeMacros.cmake]: Ditto, please use spaces.

File: media/themes/CMakeLists.txt (hunk 8 above)
Comment (binarymaster) [reply media/themes/CMakeLists.txt]: Ah, alright then.

File: media/themes/CMakeLists.txt [hunk 10]
```diff
@@ -1,3 +1,4 @@
 
 add_subdirectory(blackshade.msstyles)
 add_subdirectory(lautus.msstyles)
+add_subdirectory(modern.msstyles)
```
[the remark this answers is not in the document]
Comment (p01arst0rm) [reply media/themes/CMakeLists.txt]: already added in the newest version

File: media/themes/modern.msstyles/LICENSE [hunk 11]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
+                       Version 3, 29 June 2007
```
[the remark this answers is not in the document]
Comment (p01arst0rm) [reply media/themes/modern.msstyles/LICENSE]: this is my component, not ReactOS'. i am licensing my work to ReactOS, so, why not.

File: media/themes/modern.msstyles/lang/ru-RU.rc [hunk 12]
```diff
@@ -0,0 +1,39 @@
+LANGUAGE LANG_RUSSIAN, SUBLANG_DEFAULT
+
+STRINGTABLE
+{
+1000, "светлый"
+1001, "черный"
```
[the remark this answers is not in the document]
Comment (p01arst0rm) [reply media/themes/modern.msstyles/lang/ru-RU.rc]: as a replacement for  1000 or 1001

File: media/themes/modern.msstyles/lang/ru-RU.rc [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+STRINGTABLE
+{
+2000, "Modern светлый"
+2001, "Modern черный"
```
[the remark this answers is not in the document]
Comment (p01arst0rm) [reply media/themes/modern.msstyles/lang/ru-RU.rc]: as a replacement for 2000 or 2001

File: media/themes/modern.msstyles/lang/ru-RU.rc (hunk 12 above)
[the remark this answers is not in the document]
Comment (binarymaster) [reply media/themes/modern.msstyles/lang/ru-RU.rc]: The line number where I posted a comment is 6, so... 1001

File: media/themes/modern.msstyles/lang/zh-CN.rc [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+STRINGTABLE
+{
+1000, "亚白"
+1001, "黑暗"
+}
+
+STRINGTABLE
+{
+2000, "???"
+2001, "???"
```
Comment (binarymaster) [inline media/themes/modern.msstyles/lang/zh-CN.rc]: Something broken here?

Comment (p01arst0rm) [reply media/themes/modern.msstyles/lang/zh-CN.rc]: i dont speak chinese so i cant translate it

Comment (learn-more) [reply media/themes/modern.msstyles/lang/zh-CN.rc]: In that case please just enter the english text

**Comment by gonzoMD** [comment]
The resource files still list Pisarz as author in one string.

File: media/themes/modern.msstyles/lang/ru-RU.rc (hunk 13 above)
[the remark this answers is not in the document]
Comment (cernodile) [reply media/themes/modern.msstyles/lang/ru-RU.rc]: Comment on line 12, so 2001.

File: media/themes/modern.msstyles/lang/en-GB.rc [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+4001, "Large Fonts"
+4002, "Extra Large Fonts"
+}
+
+STRINGTABLE
+{
+5000, "Modern"
+5001, "Visual Style"
+5002, "ReactOS Foundation"
+5003, "Polar, based on win10"
```
Comment (jekader) [inline media/themes/modern.msstyles/lang/en-GB.rc]: So is this theme called Polar or Modern after all? Also, saying it's "based on win10" may be misunderstood - maybe this should be worded better, for example "inspired by win10"

Comment (p01arst0rm) [reply media/themes/modern.msstyles/lang/en-GB.rc]: i was told this was the author string in a previous comment

**Comment by learn-more** [comment]
@p01arst0rm Do you want to continue on this theme?
If so, we can commit this PR, and you can continue from there.

**Comment by p01arst0rm** [comment]
@learn-more this will be continued on when i have more free time. feel free to commit the current version.

**Comment by p01arst0rm** [comment]
the changes made are required to build the theme along with the os. there may be some other changes i have missed but that seems to get it to compile correctly.

**Comment by p01arst0rm** [comment]
thanks for the merge.

**Comment by Extravert-ir** [comment]
Hey @p01arst0rm!
Are you going continue working on the theme in some near time? If not, I think we have to pull it from source code - it looks quite unfinished compared to other themes

**Comment by p01arst0rm** [comment]
@Extravert-ir what exactly is it about this theme that looks "unfinished" ?

**Comment by jekader** [comment]
That sounds strange - if code is merged already then it should be "good enough" for use within ReactOS. And it's the responsibility of people adding new UI features to update all existing themes to support their changes.

Also in regards of workflow - if there is any functionality missing from this particular theme, a new issue should be opened to ensure the whole community sees it and knows how exactly one could help improve the situation. Commenting on a PR that has been closed 2 years ago regarding dropping code from the project sounds like the wrong way to do this.

So my suggestion would be to formulate what exactly is currently broken in the Modern theme and open an issue for that. If there is no fix and the issue is blocking further improvements or releases then it can be decided that the theme is due for removal. This will make the process more transparent.
